com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -370,30 +371,44 @@ private void OnDestroy()
             ReplNetworkState.OnValueChanged -= OnNetworkStateChanged;
         }
 
+        [ServerRpc(Delivery = RpcDelivery.Reliable)]
+        private void SubmitNetworkStateServerRpc(NetworkState networkState)
+        {
+            // as a server, apply whatever networkstate owner client sent to us, to make NetworkTransform move locally on the server
+            ApplyNetworkState(networkState);
+
+            // as a server, update netvar<networkstate> to cause it to be replicated down to clients
+            UpdateNetworkVariable(ref networkState);
+        }
+
         private void FixedUpdate()
         {
             if (!NetworkObject.IsSpawned)
             {
                 return;
             }
 
-            if (IsServer)
+            // if I'm the owner client
+            if (NetworkObject.IsOwner)
             {
-                // try to update local NetworkState
                 if (UpdateNetworkState(ref LocalNetworkState))
                 {
-                    // if updated (dirty), change NetVar, mark it dirty
-                    ReplNetworkState.Value = LocalNetworkState;
-                    ReplNetworkState.SetDirty(true);
+                    if (IsServer)
+                    {
+                        UpdateNetworkVariable(ref LocalNetworkState);
+                    }
+                    else
+                    {
+                        SubmitNetworkStateServerRpc(LocalNetworkState);
+                    }
                 }
             }
-            // try to update previously consumed NetworkState
-            // if we have any changes, that means made some updates locally
-            // we apply the latest ReplNetworkState again to revert our changes
-            else if (UpdateNetworkState(ref PrevNetworkState))
-            {
-                ApplyNetworkState(ReplNetworkState.Value);
-            }
+        }
+
+        private void UpdateNetworkVariable(ref NetworkState networkState)
```

Comment:
no no no no, not by ref! we don't want this method to touch the `NetworkState` passed in accidentally. we only want a copy of it to slap it to `ReplNetworkState`

Reply:
`ref` prevents a copy of the struct, I'd expect this to be faster, no? Dealing with not modifying the structure accidentally is easy: Let's just not modify it. :-)

Reply:
we're talking about `sizeof(NetworkState) == 40` at max :)
I'd prefer a small struct copy over leaving door open for unwanted internal state modification.